File: day1/day1.js
const fs = require("fs");
const file = fs.readFileSync("input", "utf-8");

const out = file
  .split("\n")
  .map((v) => +v)
  //   .map((val, index, arr) => {
  //     if (index > 0) {
  //       if (val > arr[index - 1]) {
  //         return { val, depth: "increase" };
  //       }
  //       return { val, depth: "decrease" };
  //     } else {
  //       return { val, depth: "n/a" };
  //     }
  //   });
  .reduce(
    (acc, val) =>
      val > acc.val
        ? { val, increments: acc.increments + 1 }
        : { val, increments: acc.increments },
    { val: 0, increments: 0 }
  );

console.log(out.increments - 1);
//console.log(out);
// console.log(
//   out.reduce((acc, entry) => {
//     entry.depth === "increase" ? acc + 1 : acc;
//   }, 0)
// );

File: day1/day1.exs
check = fn val, acc ->
  num = String.to_integer(val)

  if acc.val < num do
    %{val: num, increments: acc.increments + 1}
  else
    %{val: num, increments: acc.increments}
  end
end

getIncrements = fn %{val: _val, increments: increments} ->
  increments - 1
end

calculate = fn content ->
  content
  |> String.split("\n", trim: true)
  |> Enum.reduce(%{val: 0, increments: 0}, check)
  |> getIncrements.()
  |> IO.puts()
end

case File.read("input") do
  {:ok, content} -> calculate.(content)
  {:error} -> IO.puts("error reading file")
end

IO.puts("done")
